src/test/java/org/cbioportal/cdd/ClinicalDataDictionaryTest.java

```diff
@@ -95,6 +99,45 @@ public void getClinicalAttributeMetadataFilteredTest() throws Exception {
         assertThat(response.getBody(), containsString("{\"column_header\":\"AGE\",\"display_name\":\"Diagnosis Age\",\"description\":\"Age at which a condition or disease was first diagnosed.\",\"datatype\":\"NUMBER\",\"attribute_type\":\"PATIENT\",\"priority\":\"1\"}"));
     }
 
+    @Test
+    public void validateCacheTest() throws Exception {
+        Calendar currentDate = Calendar.getInstance();
+
+        // test condition where cache is not stale
+        // set cache age to a valid age (current date - MAXIMUM_CACHE_AGE_IN_DAYS + 1)
+        // cache is not stale so resetCache should not be called -- cache age should remain the smae
+        currentDate.add(Calendar.DATE, -(clinicalAttributesCache.MAXIMUM_CACHE_AGE_IN_DAYS - 1));
+        Date validDateOfLastCacheRefresh = currentDate.getTime();
+        clinicalAttributesCache.setDateOfLastCacheRefresh(validDateOfLastCacheRefresh);
+        if (clinicalAttributesCache.cacheIsStale()) {
+            clinicalAttributesCache.resetCache();
+        }
+        assertThat(clinicalAttributesCache.getDateOfLastCacheRefresh().toString(), equalTo(validDateOfLastCacheRefresh.toString()));
+ 
+        // test condition where cache is stale
+        // set cache age to a valid age (current date - MAXIMUM_CACHE_AGE_IN_DAYS - 3)
+        // cache is stale so resetCache should be called -- cache age should change
+        currentDate.add(Calendar.DATE, -(clinicalAttributesCache.MAXIMUM_CACHE_AGE_IN_DAYS + 3));
+        Date expiredDateOfLastCacheRefresh = currentDate.getTime();
+        clinicalAttributesCache.setDateOfLastCacheRefresh(expiredDateOfLastCacheRefresh);
+        if (clinicalAttributesCache.cacheIsStale()) {
+            clinicalAttributesCache.resetCache();
+        }
+        assertThat(clinicalAttributesCache.getDateOfLastCacheRefresh().toString(), not(equalTo(validDateOfLastCacheRefresh.toString())));
+ 
+        // resetting cache with a broken repository should throw a FailedCacheRefreshException 
```

Comment:
This is a good test, but should probably be pulled out into another test method annotated with @Test(expected = FailedCacheRefreshException.class), then you don't need the try/catch block.